src/main/java/uk/gov/ons/ctp/response/sample/scheduled/distribution/SampleUnitDistributor.java

```diff
@@ -1,170 +1,121 @@
+      } catch (InterruptedException e) {
+        // Ignored
+      }
+    }
+  }
 
-    int successes = 0;
-    int failures = 0;
+  private void processJob(CollectionExerciseJob job, RLock lock) {
     try {
-      List<CollectionExerciseJob> jobs = collectionExerciseJobRepository.findAll();
-      for (CollectionExerciseJob job : jobs) {
-        List<SampleUnit> sampleUnits;
-
-        List<Integer> excludedCases =
-            sampleUnitDistributionListManager.findList(SAMPLEUNIT_DISTRIBUTOR_LIST_ID, false);
-        log.debug("retrieve sample units excluding {}", excludedCases);
-        if (excludedCases.size() == 0) {
-          excludedCases.add(E);
-        }
+      List<uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit> mappedSampleUnits =
+          getMappedSampleUnits(job.getSampleSummaryId(), job.getCollectionExerciseId().toString());
 
-        sampleUnits =
-            sampleUnitRepository.getSampleUnits(
-                job.getSampleSummaryId(),
-                SampleSummaryDTO.SampleState.ACTIVE.toString(),
-                appConfig.getSampleUnitDistribution().getRetrievalMax(),
-                excludedCases);
-
-        if (sampleUnits.size() > 0) {
-          sampleUnitDistributionListManager.saveList(
-              SAMPLEUNIT_DISTRIBUTOR_LIST_ID,
-              sampleUnits.stream().map(SampleUnit::getSampleUnitPK).collect(Collectors.toList()),
-              true);
-        }
+      boolean hasErrors = false;
 
-        for (SampleUnit sampleUnit : sampleUnits) {
-          try {
-            uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit mappedSampleUnit =
-                mapperFacade.map(
-                    sampleUnit, uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit.class);
-            uk.gov.ons.ctp.response.sample.domain.model.SampleAttributes sampleAttributes =
-                sampleAttributesRepository.findOne(sampleUnit.getId());
-            if (sampleUnit.getId() != null) {
-              mappedSampleUnit.setId(sampleUnit.getId().toString());
-            }
-            if (sampleAttributes != null) {
-              mappedSampleUnit.setSampleAttributes(mapSampleAttributes(sampleAttributes));
-            }
-            mappedSampleUnit.setCollectionExerciseId(job.getCollectionExerciseId().toString());
-            sendSampleUnitToCollectionExerciseQueue(sampleUnit, mappedSampleUnit);
-
-            successes++;
-          } catch (CTPException e) {
-            // single case/questionnaire db changes rolled back
-            log.error(
-                "Exception {} thrown processing sampleunit {}. Processing postponed",
-                e.getMessage(),
-                sampleUnit.getSampleUnitPK());
-            log.error("Stack trace: " + e);
-            failures++;
-          }
-        }
-        sampleUnitDistributionListManager.deleteList(SAMPLEUNIT_DISTRIBUTOR_LIST_ID, true);
+      for (uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit msu : mappedSampleUnits) {
```

Comment:
Might be wrong but the import of uk.gov.ons.ctp.response.sampleunit.definition.SampleUnit it s already on the top class , i rememebr we had a conversation about it but not sure about the outcome. I think you should remove it (?)